spec.html

```diff
@@ -26609,8 +26609,8 @@ <h1>The Number Constructor</h1>
         <h1>Number ( _value_ )</h1>
         <p>When `Number` is called with argument _value_, the following steps are taken:</p>
         <emu-alg>
-          1. If no arguments were passed to this function invocation, let _n_ be *+0*.
-          1. Else, let _n_ be ? ToNumber(_value_).
+          1. If _value_ is present, let _n_ be ? ToNumber(_value_).
```

Comment:
I don't think it's odd. It's a mandatory argument (`Number.length` is `1`), and `Number()` is different than `Number(void 0)`.

Reply:
> It's a mandatory parameter (Number.length is 1),

It could be made an 'optional' parameter and Number.length could still be defined to be 1.

> and `Number()` is different than `Number(void 0).`

Agreed. But that just means that the algorithm has to distinguish between "not present" and `*undefined*`, which I'm not questioning.

> Checking the presence of a mandatory parameter is the "otherwise specified".

Agreed. So it's allowed, but it's still odd. Or maybe I'll downgrade to "unusual". (I.e., I think `Number()` and `String()` are the only built-ins where the spec checks for the presence of a 'mandatory' parameter.)

When you say "it's necessary", are you saying that the presence of the "unless otherwise specified" phrase *necessitates* that there be some place in the spec where it *is* otherwise specified? Because I'm pretty sure there are several "unless otherwise specified" sentences where it never is otherwise specified.

---

Here's a thought experiment: if the spec explicitly gave the value of the `length` property for every built-in function, would there be any useful distinction between 'optional' and 'mandatory' arguments? (I had *thought* that one distinction was that optionals didn't automatically get undefined-for-missing, and so would need an explicit 'present' test, but apparently that's not the case in practice.)

Reply:
@jmdyck currently, the only explicit lengths are for where it differs from "the number of non-optional params in the signature", all of which are legacy.

imo every argument should be implicitly undefined when omitted, and may separately be present or absent.

Reply:
> @jmdyck currently, the only explicit lengths are for where it differs from "the number of non-optional params in the signature",

(That's not quite true. E.g., the `length` of the `Array` constructor is given explicitly as 1, which is the same as its "default argument count". Similarly, the `length` of the `%TypedArray%` constructor is explicitly given as 0.)

> all of which are legacy.

Well, `Number()` and `String()` have both been around since ES1, so surely they qualify as legacy.

> imo every argument should be implicitly undefined when omitted, and may separately be present or absent.

Right, so to that extent, you're not drawing a distinction between 'optional' and 'required' parameters.